### PR TITLE
http: further header map cleanups

### PR DESCRIPTION
**Pull request by mattklein123**
- Move more headers from default O(1) headers to extensions
- Fix an issue where envoy header prefix override is used
  with custom registered headers not working.

Risk Level: Low
Testing: Existing tests
Docs Changes: N/A
Release Notes: N/A

### CODE REVIEW

**Comment by mattklein123** [comment]
@yanavlasov this will fix the issue you reported when using prefix headers. I folded this fix into a larger change that I already had going. If you prefer me to split it out I can do that.

File: source/extensions/filters/http/jwt_authn/filter.cc [hunk 1]
```diff
@@ -17,12 +17,14 @@ namespace JwtAuthn {
 namespace {
 
 Http::RegisterCustomInlineHeader<Http::CustomInlineHeaderRegistry::Type::RequestHeaders>
-    access_control_request_method(Http::Headers::get().AccessControlRequestMethod);
+    access_control_request_method_handle(Http::CustomHeaders::get().AccessControlRequestMethod);
+Http::RegisterCustomInlineHeader<Http::CustomInlineHeaderRegistry::Type::RequestHeaders>
+    origin_handle(Http::CustomHeaders::get().AccessControlRequestMethod);
```
Comment (nareddyt) [inline source/extensions/filters/http/jwt_authn/filter.cc:22]: @mattklein123 shouldn't this be `Http::CustomHeaders::get().Origin`?

Comment (qiwzhang) [reply source/extensions/filters/http/jwt_authn/filter.cc:22]: I will send out a pr to fix this typo

Comment (qiwzhang) [reply source/extensions/filters/http/jwt_authn/filter.cc:22]: https://github.com/envoyproxy/envoy/pull/11911

**Pull request merged**